src/puppetlabs/puppetdb/cli/benchmark.clj

```diff
@@ -194,9 +194,9 @@
   [{:keys [host lastrun catalog report factset base-url run-interval rand-percentage] :as state} clock]
   (if-not (> (- clock lastrun) run-interval)
     state
-    (let [catalog (some-> catalog update-catalog (maybe-tweak-catalog rand-percentage))
-          report (some-> report update-report-run-fields)
-          factset (some-> factset (update-factset rand-percentage))]
+    (let [catalog (some->> catalog update-catalog (maybe-tweak-catalog rand-percentage))
+          report (some-> report update-report-run-fields json/generate-string)
+          factset (some->> factset (update-factset rand-percentage) json/generate-string)]
```

Comment:
not in a position to test this ATM, but won't this recursively json encode reports and factsets when the -i flag is used?